preciceconfigchecker/rule.py

```diff
@@ -1,37 +1,142 @@
-from enum import Enum
+from abc import ABC, abstractmethod
+from typing import List
+from severity import Severity
 
 
-class Severity(Enum):
-    INFO = "Info"
-    WARNING = "Warning"
+class Rule(ABC):
+    """
+    Abstract Class 'Rule'. Checking a 'Rule' for violations and producing formatted output.
+    """
 
+    violation_counter:int = 0
+    """Static Attribute: Do not use 'self.numbers', use 'Rule.numbers'"""
 
-class Rule:
-    severity:Severity
-    name:str
-    message:str
-    possible_solutions:str
-    followed:bool = True
-    numbers_not_followed:int = 0
+    def __init__(self, severity:Severity, problem:str) -> None:
+        """
+        Initializes an Rule object.
 
-    def __init__(self, severity:Severity = Severity.INFO, name:str = "Rule", message:str = "", possible_solutions:str = "") -> None:
+        Args:
+            severity (Severity): Type
+            problem (str): Short explanation of what the rule is supposed to check in general.
```

Comment:
does this make sense here? why not have a function that prints information about the rule? like print_info(): "Checks whether or not a node has an outgoing edge" 
especially when we have a rule class per issue, then initializing DataNotUsedRule with (...,"Checks whether data is not used") seems redundant to me

Reply:
The idea behind it: If you have several violations per rule, a short explanation at the beginning is helpful to understand the following violations more quickly. We can describe each individual violation in as much detail as we want.

Reply:
If we only expect a small number of violations, or if the violations themselves briefly describe the problem, we can also make this parameter optional.

Reply:
Let's solve this in the other PR. #14